File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hash_map = {} 
        output = [] 
        
        for i in range(len(nums)):
            value = target - nums[i] 
            
            if value in hash_map:
                output.append(hash_map[value])
                output.append(i)
                
                return output 
            
            else: 
                hash_map[nums[i]] = i 
                
                
        
                
                
            
        
        

File: 0001-two-sum/0001-two-sum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {

        int[] output = new int[2];
        HashMap<Integer, Integer> hashMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (hashMap.containsKey(target - nums[i])) {
                int index = hashMap.get(target - nums[i]); 
                output[0] = index;
                output[1] = i; 
                return output; 
                
            } else {
                hashMap.put(nums[i], i); 
            }
            
        }


        return output; 
    }
}